app/services/BulletPointBuilder.scala

```diff
@@ -102,28 +106,42 @@ object BulletPointBuilder {
   def locateTextsAndMatchesContainingLeadingText(
```

Comment:
This fails the > 50 lines test. Its a difficult fn to follow, the boolean switch argument and the nested ifs compound situation. Splitting into two functions to remove boolean switch arg would produce two simpler fns

Reply:
Split function into two although this did mean relinquishing the tailrec annotation.